untitled1/src/com/aca/homework/week5/light/out/Game.java

```diff
@@ -0,0 +1,167 @@
+package com.aca.homework.week5.light.out;
+
+import java.util.Scanner;
+
+public class Game {
+    private Light[][] lights;
+    private int moves;
+
+    public Game() {
+        initialize();
+    }
+
+    private void initialize() {
+        lights = new Light[5][5];
+        for (int i = 0; i < lights.length; i++) {
+            for (int j = 0; j < lights[i].length; j++) {
+                if (isAcceptCertainValues(i, j)) {
+                    lights[i][j] = new Light(false);
+                } else
+                    lights[i][j] = new Light(true);
+            }
+        }
+    }
+
+    private void start() {
+        String coordinate = inputCoordinate();
+        switchStates(Character.getNumericValue(coordinate.charAt(1)), Character.getNumericValue(coordinate.charAt(3)));
+    }
+
+    private boolean isWin() {
+        for (int i = 0; i < lights.length; i++) {
+            for (int j = 0; j < lights[i].length; j++) {
+                if (lights[i][j].getLight() == '+') {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private void switchStates(int i, int j) {
+        lights[i][j].switchState();
+        if (isValidCoordinate(i + 1, j)) lights[i + 1][j].switchState();
+        if (isValidCoordinate(i - 1, j)) lights[i - 1][j].switchState();
+        if (isValidCoordinate(i, j + 1)) lights[i][j + 1].switchState();
+        if (isValidCoordinate(i, j - 1)) lights[i][j - 1].switchState();
+    }
+
+    private String inputCoordinate() {
```

Comment:
this can return an array of coordinates or even the Point or Coordinate type. is it not a good practice to return a string